src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py

```diff
@@ -3720,9 +3724,13 @@
     short-summary: Space-separated list of services allowed private access to this subnet.
     populator-commands:
       - az network vnet list-endpoint-services
+  - name: --nat-gateway
+    short-summary: Attach Nat Gateway to subnet
 examples:
   - name: Associate a network security group to a subnet.
     text: az network vnet subnet update -g MyResourceGroup -n MySubnet --vnet-name MyVNet --network-security-group MyNsg
+  - name: Update subnet with Nat Gateway
+    text: az network vnet subnet update -n MySubnet --vnet-name MyVnet -g MyResourceGroup --nat-gateway nat_gateway_test --address-prefixes "10.0.0.0/21"
```

Comment:
Please use "natGatewayTest" to be consistent with the other values in the example.